eppoclient/client_test.go

```diff
@@ -170,42 +170,75 @@ func Test_AssignSubjectWithAttributesAndRules(t *testing.T) {
 	client := newEppoClient(mockConfigRequestor, mockLogger)
 
 	for _, tt := range tests {
-		assignment, _ := client.GetAssignment(tt.a, tt.b, tt.c)
+		assignment, _ := client.GetStringAssignment(tt.a, tt.b, tt.c)
 
 		assert.Equal(t, tt.want, assignment)
 	}
 }
 
 func Test_WithSubjectInOverrides(t *testing.T) {
-	var mockLogger = new(mockLogger)
-	mockLogger.Mock.On("LogAssignment", mock.Anything).Return()
-
-	var mockConfigRequestor = new(mockConfigRequestor)
-	var mockVariations = []Variation{
-		{Name: "control", ShardRange: shardRange{Start: 0, End: 100}},
-	}
-	var overrides = make(dictionary)
-	overrides["d6d7705392bc7af633328bea8c4c6904"] = "override-variation"
-	var allocations = make(map[string]Allocation)
-	allocations[defaultAllocationKey] = Allocation{
-		PercentExposure: 1,
-		Variations:      mockVariations,
-	}
-	var mockResult = experimentConfiguration{
-		Name:          "recommendation_algo",
-		Enabled:       true,
-		SubjectShards: 1000,
-		Overrides:     overrides,
-		Rules:         []rule{textRule},
+	var tests = []struct {
+		name               string
+		input              Value
+		inputUnwantedValue Value
+		inputValueType     ValueType
+		want               Value
+	}{
+		{"string override", String("override-variation"), String("foo"), StringType, String("override-variation")},
+		{"numeric override", Numeric(5), Numeric(100), NumericType, Numeric(5)},
+		{"boolean override", Bool(true), Bool(false), BoolType, Bool(true)},
 	}
 
-	mockConfigRequestor.Mock.On("GetConfiguration", "experiment-key-1").Return(mockResult, nil)
-
-	client := newEppoClient(mockConfigRequestor, mockLogger)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			var mockLogger = new(mockLogger)
+			mockLogger.Mock.On("LogAssignment", mock.Anything).Return()
+
+			var mockConfigRequestor = new(mockConfigRequestor)
+			var mockVariations = []Variation{
+				{Name: "control", Value: tt.inputUnwantedValue, ShardRange: shardRange{Start: 0, End: 100}},
```

Comment:
the `inputUnwantedValue` is over-written

Reply:
for example:

```
				assignment, _ := client.GetStringAssignment("user-1", "experiment-key-1", dictionary{})
				if assignment != tt.want.stringValue {
					t.Errorf("got %s, want %s", assignment, tt.want.stringValue)
				}
				if assignment == tt.inputUnwantedValue.stringValue {
					t.Errorf("assignment should not have un-overwritten value", assignment, tt.inputUnwantedValue.stringValue)
				}
```

Reply:
cool sounds good, now that I'm looking at it again just comparing to the expected value is sufficient and I've removed all this in https://github.com/Eppo-exp/golang-sdk/pull/22/commits/16546d1d5abd9ed4ad880662a83660a9d9876dc5